File: api/src/main/java/org/delivery/api/common/api/Result.java
package org.delivery.api.common.api;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.delibery.common.error.ErrorCode;
import org.delibery.common.error.ErrorCodeIfs;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Result {

    private Integer resultCode;

    private String resultMessage;

    private String resultDescription;

    public static Result OK() {
        return Result.builder()
                .resultCode(ErrorCode.OK.getErrorCode())
                .resultMessage(ErrorCode.OK.getDescription())
                .resultDescription("성공")
                .build();
    }

    public static Result ERROR(ErrorCodeIfs errorcodeIfs) {
        return Result.builder()
                .resultCode(errorcodeIfs.getErrorCode())
                .resultMessage(errorcodeIfs.getDescription())
                .resultDescription("에러 발생")
                .build();
    }

    public static Result ERROR(ErrorCodeIfs errorcodeIfs, Throwable tx) {
        return Result.builder()
                .resultCode(errorcodeIfs.getErrorCode())
                .resultMessage(errorcodeIfs.getDescription())
                .resultDescription(tx.getLocalizedMessage())
                .build();
    }

    public static Result ERROR(ErrorCodeIfs errorcodeIfs, String description) {
        return Result.builder()
                .resultCode(errorcodeIfs.getErrorCode())
                .resultMessage(errorcodeIfs.getDescription())
                .resultDescription(description)
                .build();
    }
}

File: service/common/src/main/kotlin/org/delivery/common/api/Result.kt
package org.delivery.common.api

import org.delivery.common.error.ErrorCode
import org.delivery.common.error.ErrorCodeIfs

data class Result(
    val resultCode: Int ?= null,
    val resultMessage: String ?= null,
    val resultDescription: String ?= null,
) {
    companion object { // static 함수와 동일하게 동작
        @JvmStatic
        fun OK(): Result {
            return Result(
                resultCode = ErrorCode.OK.getErrorCode(),
                resultMessage = ErrorCode.OK.getDescription(),
                resultDescription = "성공"
            )
        }

        @JvmStatic
        fun ERROR(errorCodeIfs: ErrorCodeIfs): Result {
            return Result(
                resultCode = errorCodeIfs.getErrorCode(),
                resultMessage = errorCodeIfs.getDescription(),
                resultDescription = "에러 발생"
            )
        }

        @JvmStatic
        fun ERROR(errorCodeIfs: ErrorCodeIfs, tx: Throwable): Result {
            return Result(
                resultCode = errorCodeIfs.getErrorCode(),
                resultMessage = errorCodeIfs.getDescription(),
                resultDescription = tx.localizedMessage
            )
        }

        @JvmStatic
        fun ERROR(errorCodeIfs: ErrorCodeIfs, description: String): Result {
            return Result(
                resultCode = errorCodeIfs.getErrorCode(),
                resultMessage = errorCodeIfs.getDescription(),
                resultDescription = description
            )
        }
    }
}